### DISCUSSION TITLE
Updating store no longer take effect immediately on 1.4.x

### DISCUSSION THREAD

**Discussion by weijarz** [solidjs/solid#1038] [Q&A] [answered]
[Playground](<https://playground.solidjs.com/?version=1.4.1#NobwRAdghgtgpmAXGGUCWEwBowBcCeADgsrgM4Ae2YZA9gK4BOAxiWGjIbY7gAQi9GcCABM4jXgF9eAM0a0YvADo1aAGzQiAtACsyAegDucAEYqA3EogcuPfr1oQAsgwh9pchbwDkdDdr1vKxtuPgFmIShcOABlXG44KVl5RV91TV0DMnihIIgrOApbPjEZKHo1Phl6CGZcNEdeAGFXaMYACgBKfiteXmZHbN5gbISsXjI4XDiEgF1eAF5+yOiZoXaBNDIAGVooEQwAc0RZKDVJqU6rXv7BvjU9kUXeLsWAPh6IPr7J6Zy4Da8La7fZHE64Rj0RKSK5fb5oGQvACEoyEADpgY8jt1cAALeSGXgQOCEgCijHkHW8pIASjSAPI0pHeWF9STXOGOFw1XDtV4LD4CB77V4wm76fS8YUiLocvpCXBML4AHlxAEY3iBUXAMTssRBDrwAPw+EEHA3eXgnbwAOVouDNR28kmV+nVbys7PyECEonEfO6At4ypaPPEvH0b3GIlozHo8DcaMOU1JajgCdwACF8ABJGUqKCEQgqTpIzqWTCSWZAA>)

```ts
import { render } from "solid-js/web";
import { onMount } from 'solid-js'
import { createStore } from 'solid-js/store'

export default function Counter() {
  const [store, setStore] = createStore({ isLoading: false })

  const load = () => {
    setStore({ isLoading: true })
    if (!store.isLoading) throw new Error('ERROR!')
  }

  onMount(() => { load() })
  // load()

  return <h1>{store.isLoading ? 'Loading' : 'NotLoading'}</h1>
}

render(() => <Counter />, document.getElementById("app")!);
```

After I upgraded the solid-js to the latest version (1.4.x), my app no longer works.
I found that the reason is that updating the store does not take effect immediately.
```ts
onMount(() => {
  setStore({ isLoading: true })
  console.log(store.isLoading)  // Here store.isLoading is still false
})
```

Is it a feature or bug?

**Comment by ryansolid** [comment, marked as answer]
Feature. Other Signals have always worked this way but Stores were inconsistent. To be clear this is a behavior around batching effects which is under review. But I prioritized making all reactivity consistent.
